src/Forms/GridField/GridField.php

```diff
@@ -444,6 +451,66 @@ private function initState(): void
         }
     }
 
+    /**
+     * Adds state for this gridfield from the request variables.
+     *
+     * If there is state already set on this GridField, that takes precedent
+     * over state from the request.
+     */
+    private function addStateFromRequest(): void
+    {
+        $request = $this->getRequest();
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+        
+        $stateStr = $this->getStateManager()->getStateFromRequest($this, $request);
+
+        if ($stateStr) {
+            $oldState = $this->getState(false);
+            // Create a dummy state so that we can merge the current state with the request state.
+            $newState = new GridState($this, $stateStr);
+            // Put the current state on top of the request state.
+            $newState->setValue($oldState->Value());
+            $this->state = $newState;
+        }
+    }
+
+    /**
+     * Add get and post parameters if exist to the URL.
+     * URL keeps state for GridField, that exist in few different tabs.
+     */
+    public function addRequestToURL(string $link): string
+    {
+        $request = $this->getRequest();
+
+        if (($request instanceof NullHTTPRequest) && Controller::has_curr()) {
+            $request = Controller::curr()->getRequest();
+        }
+
+        $postparams = [];
+
+        if ($request->postVars()) {
+            foreach ($request->postVars() as $key => $val) {
+                if (stripos($key, 'gridState') === 0) {
+                    $postparams[$key] = $val;
+                }
+            }
+        }
+
+        $params = array_merge($request->getVars(), $postparams);
+
+        if ($params && is_array($params) && count($params) > 0) {
+            foreach ($params as $key => $val) {
+                $link = HTTP::setGetVar($key, $val, $link);
+            }
+        }
```

Comment:
You should make sure you're not including this gridfield's own state here - so `if ($key === $this->getStateManager()->getStateKey($this))` then don't add it.

Reply:
I'm not sure that we should skip current State if we have this GridField open and we should apply filtering and sorting. Could you explain, please.

Reply:
The state for the _current_ gridfield is handled by the state manager already. So if we keep that here, we're duplicating the effort that state manager is already doing.
It seems to me that the point of getting data from the request here is to get state for _other_ gridfields, right?

Reply:
DONE